app/models/cart.rb

```diff
@@ -46,8 +46,10 @@ def add_quantity(quantity, item, inventory)
   end
 
   def remove_quantity(quantity, item)
-    quantity.times do
-      remove_item(item.id.to_s)
+    if @contents.values.first > 0
```

Comment:
Added the if statement so you cant decrease quantity if you are already at 0.

Reply:
looks good